File: solutions/0049-group-anagrams/solution.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>>m;
        for(auto s:strs){
            string sc = s;
            sort(sc.begin(),sc.end());
            m[sc].push_back(s);
        }
        vector<vector<string>>ans;

        for(auto [key,value] : m){
            ans.push_back(value);
        }
        return ans;
    }
};

File: solutions/0049-group-anagrams/solution.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        m = {}
        for s in strs:
            key = "".join(sorted(s))
            if key in m:
                m[key].append(s)
            else:
                m[key]=[s]
        ans = []
        for k in m:
            ans.append(m[k])
        return ans


        
